src/librustdoc/html/render/mod.rs

```diff
@@ -3517,14 +3516,17 @@ fn render_assoc_items(
                 RenderMode::Normal
             }
             AssocItemRender::DerefFor { trait_, type_, deref_mut_ } => {
+                let id = small_url_encode(&format!("deref-methods-target-{:#}", type_.print()));
```

Comment:
Also I just realized: I seem to recall us having a hashset to prevent IDs duplication. Maybe they should use it too. That could also allow to have shorter IDs.

Reply:
`target` is just here as a prefix matching the text of section name, as in "Target = Foo". It could easily be removed as well, no strong opinion from my side.

I'll look into these suggestions for a more stable ID.

Reply:
Okay, I removed the `target` prefix from the ID. I also changed to use the same `derive_id` uniqueness code as in other places. Since we need to access those same derived IDs in the sidebar rendering path, I added a new map to the `Context` to manage this.